controllers/src/main/scala/cromwell/pipeline/controller/AkkaHttpClient.scala

```diff
@@ -4,67 +4,113 @@ import akka.actor.ActorSystem
 import akka.http.scaladsl.Http
 import akka.http.scaladsl.model.Uri.Query
 import akka.http.scaladsl.model.headers.RawHeader
-import akka.http.scaladsl.model.{ HttpMethods, HttpRequest, HttpResponse, Uri }
+import akka.http.scaladsl.model.{ HttpEntity, HttpMethods, HttpRequest, HttpResponse, Uri }
 import akka.stream.ActorMaterializer
-import cromwell.pipeline.service.{ HttpClient, Response }
+import cromwell.pipeline.service.{ HttpClient, NegativeResponseBody, PositiveResponseBody, Response }
+import play.api.libs.json.{ JsResult, JsValue, Json, Reads, Writes }
+import akka.http.scaladsl.model.StatusCodes
 
 import scala.concurrent.duration._
 import scala.concurrent.{ ExecutionContext, Future }
 
 class AkkaHttpClient(implicit actorSystem: ActorSystem, materializer: ActorMaterializer) extends HttpClient {
   private val expirationTime: FiniteDuration = 300.millis
+  private val httpContentTypeJson = "application/json"
 
-  override def get(url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
-    implicit ec: ExecutionContext
-  ): Future[Response] = {
+  override def get[B](url: String, params: Map[String, String] = Map(), headers: Map[String, String] = Map())(
+    implicit ec: ExecutionContext,
+    f: Reads[B]
+  ): Future[Response[B]] = {
     val futureResponse: Future[HttpResponse] = Http().singleRequest(
       HttpRequest(method = HttpMethods.GET, uri = Uri(url).withQuery(Query(params))).withHeaders(parseHeaders(headers))
     )
     responsify(futureResponse)
   }
 
-  private def responsify(futureResponse: Future[HttpResponse])(implicit ec: ExecutionContext): Future[Response] =
+  private def responsify[B](
+    futureResponse: Future[HttpResponse]
+  )(implicit ec: ExecutionContext, f: Reads[B]): Future[Response[B]] =
     futureResponse.flatMap { response =>
-      response.entity.toStrict(expirationTime).map { body =>
-        Response(
+      response.status match {
+        case StatusCodes.Created    => positiveResponse(response)
+        case StatusCodes.Accepted   => positiveResponse(response)
+        case StatusCodes.BadRequest => negativeResponse(response)
+        case StatusCodes.NotFound   => negativeResponse(response)
+        case _                      => negativeResponse(response)
```

Comment:
I don't think there is a need to explicitly pattern match over so many cases if we handle all of them in the same way.
```
case StatusCodes.Created | StatusCodes.Accepted => positive
case _ => negative
```

Reply:
replaced